### PR TITLE
renaming PodSelector to LabelSelector

### PR DESCRIPTION
**Pull request by sdminonne**
as discussed here https://github.com/kubernetes/kubernetes/issues/341#issuecomment-160794047
@davidopp @bgrant0607 @mikedanese @soltysh

### CODE REVIEW

**Comment by davidopp** [comment]
I'm not sure we should rename to LabelSelector, because NodeSelector will also be a form of label selector.

**Comment by soltysh** [comment]
You're missing `docs/api-reference/extensions/v1beta1/definitions.html` update. Please run `hack/update-all.sh` to update it. 

@davidopp as mentioned in the original issue I prefer the general selector type to be rather called `LabelSelector` than `PodSelector`. The latter suggests it's meant only for pods, whereas the former suggests wider adoption. Even if `NodeSelector` is a subset its name suggests its applicable only to Nodes, and that's valid assumption we should encourage, imho.

**Comment by bgrant0607** [comment]
I agree with @soltysh. I'm fine with LabelSelector.

**Comment by sdminonne** [comment]
hmm for some reasons (need to investigate)  hack/update-all.sh broke something.

**Comment by soltysh** [comment]
@sdminonne I've tried it before, it worked perfectly ok on my machine. Maybe try `make clean` before?

**Comment by sdminonne** [comment]
@soltysh sure the issue comes from my devenv. Just looking at it.

**Comment by sdminonne** [comment]
@soltysh (running scripts I lost this pkg/apis/extensions/types.generated.go)

**Comment by sdminonne** [comment]
and this pkg/apis/extensions/v1beta1/types.generated.go

**Comment by sdminonne** [comment]
it should be better now (rebased as well)

**Comment by davidopp** [comment]
I'm fine with renaming api.PodSelector to api.LabelSelector, but I'm worried about confusion with the existing labels.LabelSelector. @bgrant0607 pointed out that they're in different packages, which is true, but people tend to use "grep" and it would be nice to be able to distinguish them (especially since we will probably reuse labels.LabelSelector for NodeSelector when we implement NodeSelector).

So, could you (in this PR) also rename the existing labels.LabelSelector to labels.SelectorInternal ? (Let's wait to see what @bgrant0607 thinks of that name; it's the best I could think of off the top of my head. My second choice would be just labels.Selector)

**Comment by soltysh** [comment]
The PR so far LGTM, but I'm fine with change @davidopp proposed.

**Comment by sdminonne** [comment]
@soltysh @davidopp can it be another PR? I'll do tomorrow morning since this PR is very close to  be mergeable and modifying labels.LabelSelector to labels.InternalSelector could be quite big... And to be honest I thinkg the right name should be label.Selector (getting rid of the interface)

**Comment by mikedanese** [comment]
I don't think LabelSelector really needs to be exposed since the Selector interface is mutable. I like internalSelector and expose a constructor.

File: pkg/apis/extensions/types.go [hunk 1]
```diff
@@ -298,7 +298,7 @@ type DaemonSetSpec struct {
 	// Must match in order to be controlled.
 	// If empty, defaulted to labels on Pod template.
```
Comment (mikedanese) [inline pkg/apis/extensions/types.go]: Bad doc

File: pkg/apis/extensions/types.go [hunk 2]
```diff
@@ -403,7 +403,7 @@ type JobSpec struct {
 	Completions *int `json:"completions,omitempty"`
 
 	// Selector is a label query over pods that should match the pod count.
```
Comment (mikedanese) [inline pkg/apis/extensions/types.go]: Bad doc

File: pkg/apis/extensions/v1beta1/types.go [hunk 3]
```diff
@@ -655,26 +655,26 @@ type ClusterAutoscalerList struct {
 	Items []ClusterAutoscaler `json:"items"`
 }
 
-// A pod selector is a label query over a set of pods. The result of matchLabels and
+// A label selector is a label query over a set of pods. The result of matchLabels and
```
Comment (mikedanese) [inline pkg/apis/extensions/v1beta1/types.go]: Bad doc

**Comment by mikedanese** [comment]
LGTM after documentation is fixed

**Comment by sdminonne** [comment]
@mikedanese doc (in .../types.go) fixed (squashed).
Tomorrow (my tomorrow) if nothing change I'll submit another PR renaming labels.LabelSelector to labels.internalSelector adding a constructor (labels.NewSelector). Let me know.

**Comment by ncdc** [comment]
@k8s-bot unit test this

**Comment by ncdc** [comment]
cc @kubernetes/rh-cluster-infra

**Comment by mikedanese** [comment]
This is LGTM once ci is passing. Looks like the codecgen is messed up again. Mind rebasing and fixing?

**Comment by sdminonne** [comment]
rebased and squashed

**Comment by mikedanese** [comment]
LGTM

**Comment by sdminonne** [comment]
@gmarek @wojtek-t may you help me to get this merged (if I rebase the lgtm label will disappear...)?

**Comment by wojtek-t** [comment]
@sdminonne - we have a huge backlog of lgtm-ed PRs now, so it may take some time
However, I'm happy to re-add lgtm label after you rebase it

**Comment by sdminonne** [comment]
trying again... squashed. 
@wojtek-t thanks!

**Comment by gmarek** [comment]
@sdminonne: lot of failures - it's unlikely a flake, but I'll rerun the test to be sure.
```
Test Result (10 failures / +10)
Kubernetes e2e suite.Kubectl client Simple pod should support exec
Kubernetes e2e suite.hostPath should give a volume the correct mode [Conformance]
Kubernetes e2e suite.Kubectl client Update Demo should scale a replication controller [Conformance]
Kubernetes e2e suite.EmptyDir volumes should support (root,0777,default) [Conformance]
Kubernetes e2e suite.Downward API volume should update annotations on modification [Conformance]
Kubernetes e2e suite.Daemon set should run and stop simple daemon
Kubernetes e2e suite.Kubectl client Simple pod should support exec through an HTTP proxy
Kubernetes e2e suite.ReplicationController should serve a basic image on each replica with a public image [Conformance]
Kubernetes e2e suite.Kubectl client Update Demo should do a rolling update of a replication controller [Conformance]
Kubernetes e2e suite.Services should be able to create a functioning NodePort service
```

@k8s-bot test this

**Comment by sdminonne** [comment]
@gmarek thanks!

**Comment by gmarek** [comment]
@sdminonne it seems that it was a big flake:)

**Comment by sdminonne** [comment]
@gmarek :+1: thanks again

**Pull request merged**